lib/sgtn-client/api/base.rb

```diff
@@ -0,0 +1,92 @@
+# 
+#  Copyright 2019-2022 VMware, Inc.
+#  SPDX-License-Identifier: EPL-2.0
+#
+
+require 'multi_json'
+
+module SgtnClient
+
+  class Base
+    class << self
+
+      def getTranslation(component, key, locale)
+        locale = SgtnClient::LocaleUtil.get_best_locale(locale)
+        items = get_cs(component, locale)
+        if items.nil? || items["messages"].nil?
+          nil
+        else
+          items["messages"][key]
+        end
+      end
+
+      def compareSource(component, key, default_locale, source, translation)
+        SgtnClient.logger.debug "[Base][compareSource]component=#{component},key=#{key},default_locale=#{default_locale},source=#{source},translation=#{translation}"
+        items = get_cs(component, default_locale)
+        if items.nil? || items["messages"].nil?
+          translation
+        else
+          target = items["messages"][key]
+          source == target ? translation : source
+        end
+      end
+
+      def get_cs(component, locale)
+        cache_key = SgtnClient::CacheUtil.get_cachekey(component, locale)
+        SgtnClient.logger.debug "[Base][get_cs]cache_key=#{cache_key}"
+        expired, items = SgtnClient::CacheUtil.get_cache(cache_key)
+        if items.nil? || expired
+          items = load(component, locale)
+          if items.nil?
+            items = SgtnClient::Source.getSources(component, SgtnClient::Config.configurations.default)
+            SgtnClient::Core::Cache.put(cache_key, items, 60)
+          else
+            SgtnClient::CacheUtil.write_cache(cache_key, items)
+          end
+        else
+          SgtnClient.logger.debug "[Base]get translations from cache with key: " + cache_key
+        end
+
+        return items
+       end
+
+      def load(component, locale)
+        env = SgtnClient::Config.default_environment
+        mode = SgtnClient::Config.configurations[env]["bundle_mode"]
+        SgtnClient.logger.debug "[Base][load]mode=#{mode}"
+        if mode == 'offline'
+          return load_o(component, locale)
+        else
+          return load_s(component, locale)
+        end
+      end
+
+      def load_o(component, locale)
+        env = SgtnClient::Config.default_environment
+        product_name = SgtnClient::Config.configurations[env]["product_name"]
+        version = SgtnClient::Config.configurations[env]["version"].to_s
+        translation_bundle = SgtnClient::Config.configurations[env]["translation_bundle"]
+        bundlepath = translation_bundle + "/" + product_name + "/" + version + "/" + component + "/messages_" + locale + ".json"
+        SgtnClient::FileUtil.read_json(bundlepath)
+      end
+
+      def load_s(component, locale)
```

Comment:
Please make the name clear. Hard for others to read.

Reply:
Let's don't change it in this PR, i just move the whole block from translation.rb.

Reply:
Marking this as unresolved until it is tracked elsewhere.